lib/validation.js

```diff
@@ -67,7 +67,8 @@ function compileSchemasForValidation (context, compile) {
 }
 
 function validateParam (validatorFunction, request, paramName) {
-  const ret = validatorFunction && validatorFunction(request[paramName])
+  const isUndefined = request[paramName] === undefined
+  const ret = validatorFunction && validatorFunction(isUndefined ? null : request[paramName])
```

Comment:
I'm not 100% sure of this approach. I think it's good as we're explicitly passing `null` indicating the absence of value instead of `undefined` which only exists in JavaScript. Please, let me know your thoughts.

Reply:
What is this solving?

Reply:
Basically, if we set `undefined` as the default value for an empty req body when running the schema validation, doesn't matter if the schema allows the body to be nullable, as we're passing `undefined` the validation will throw right away. Changing from `undefined` to `null` allows those schemas which are nullable to validate and pass if the value it's empty (at least the body)

Reply:
I see. This is validating the value of path parameters. How can a path parameter be null? @Eomm what is your opinion here?

Reply:
For now yes, I can reduce it to only the `body` for sure. I'll like to hear your opinion, maybe I'm missing something where there's a better way to handle it

Reply:
This function is called 4 times with the `paramName` equals to `params`, `querystring`, `headers` and `body`
(only in the case `paramName===body` this check is needed if I have understood the case)
I saw there are some tests that cover this case already

I think this check is faster than a string compare so it LGTM